# Analyzing CIA Factbook Data Using SQL

Aim of this project is to explore & analyze demographic information. The data comes from [CIA World Factbook](https://www.cia.gov/library/publications/the-world-factbook/), a compendium of statistics about all of the countries on Earth The Factbook contains demographic information like:

- population - The population as of 2015.
- population_growth - The annual population growth rate, as a percentage.
- area - The total land and water area.


In [1]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

Let's explore first 5 rows of the table.

In [3]:
%%sql
SELECT *
    FROM facts
  LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Here are the description for some of the columns:

- name - The name of the country.
- area- The country's total area (both land and water).
- area_land - The country's land area in square kilometers.
- area_water - The country's waterarea in square kilometers.
- population - The country's population.
- population_growth- The country's population growth as a percentage.
- birth_rate - The country's birth rate, or the number of births a year per 1,000 people.
- death_rate - The country's death rate, or the number of death a year per 1,000 people.

### Summary statistics

In [6]:
%%sql
SELECT MIN(population) AS min_pop, 
        MAX(population) AS max_pop, 
        MIN(population_growth) AS min_pop_growth,
        MAX(population_growth) AS max_pop_growth
    FROM facts;

Done.


[(0, 7256490011, 0.0, 4.02)]

From the table above we can conclude that:

- some country has 0 population what we have to explore further,
- max_population of a country is 7,256,490,011 that also required further analysis. we explore that there are outlier countries

First let's explore which country has a population equal 0.

In [10]:
%%sql
SELECT *
    FROM facts
    WHERE population == (SELECT MIN(population)
                         FROM facts)

Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

From the table above we can see that population is 0 for Antarctica. Which makes sense and match the CIA Factbook [page for Antarctica](https://www.cia.gov/library/publications/the-world-factbook/geos/ay.html).

![title](Antarctica.png)

Next, let's see the row with maximum population.

In [11]:
%%sql
SELECT *
    FROM facts
    WHERE population == (SELECT MAX(population)
                        FROM facts)

Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

As we can see from the table above the row that contains max population refers to the entire world, which explains the maximum population of over 7.2 billion we found earlier.

Now we can recalculate the summary statistics calculated earlier, excluding the row for the whole world.

In [16]:
%%sql
SELECT MIN(population) AS min_pop, 
        MAX(population) AS max_pop, 
        MIN(population_growth) AS min_pop_growth,
        MAX(population_growth) AS max_pop_growth
    FROM facts
    WHERE population != (SELECT MAX(population)
                        FROM facts);

Done.


[(0, 1367485388, 0.0, 4.02)]

Max_population is 1,367,485,388 which is logic value , let's discover it's country.

In [47]:
%%sql
SELECT * 
    FROM facts
    WHERE population=(
        SELECT MAX(population) FROM facts
        WHERE name != 'World');

Done.


[(37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44)]

China have the maximum population 1,367,485,388 with population growth rate 0.45

### Population and Area

In [48]:
%%sql
-- calculate the average value for both population & area
SELECT AVG(population) AS avg_population, AVG(area) AS avg_area
    FROM facts
    WHERE name <> 'World';

Done.


[(32242666.56846473, 555093.546184739)]

From our output the average population is around 32 million whereas the average area is approximately 555 thousand Kilometer square

Now let us explore which countries are densely populated by find:

- Countries above average values for population
- and Below average values for area

In [58]:
%%sql
SELECT *
    FROM facts
  WHERE population > (SELECT AVG(population)
                           FROM facts
                          WHERE name <> 'World'
                       )
  AND area < (SELECT AVG(area)
                     FROM facts
                    WHERE name <> 'World'
                 )

Done.


[(14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (80, 'iz', 'Iraq', 438317, 437367, 950, 37056169, 2.93, 31.45, 3.77, 1.62),
 (83, 'it', 'Italy', 301340, 294140, 7200, 61855120, 0.27, 8.74, 10.19, 4.1),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (91, 'ks', 'Korea, South', 99720, 96920, 2800, 49115196, 0.14, 8.19, 6.75, 0.0),
 (120, 'mo', 'Morocco', 446550, 446300, 250, 33322699, 1.0, 18.2, 4.81, 3.36),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09),
 (139, 'pl', 'Poland', 312685, 304255, 8430, 38562189, 0.09, 9.74, 10.19, 0.46),
 (163, 'sp', 'Spain', 505370, 498980, 6390, 48146134, 0.89, 9.64, 9.04, 8.31),
 (173, 'th', 'Thailand', 513120, 510890, 2230, 67976405, 0.34, 11.19, 7.8, 0.0),
 (182, 'ug', 'Uganda', 241038, 197100, 43938, 37101745, 3.24, 43.79, 10.69, 0.74),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3)]

In conclusion, if we compare our result with the real data of some of these countries, we will find the result to be true.

### People

In [51]:
%%sql
-- Country has the most people? 
SELECT * 
    FROM facts
    WHERE population =(SELECT MAX(population) FROM facts
                      WHERE name <> 'World');

Done.


[(37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44)]

As found before China have the maximum population 1,367,485,388 with population growth rate 0.45. Let's see which country has the highest growth rate.

In [27]:
%%sql 
SELECT *
    FROM facts
 WHERE population_growth == (SELECT MAX(population_growth)
                                       FROM facts
                                       WHERE name <> 'World');

Done.


[(162, 'od', 'South Sudan', 644329, None, None, 12042910, 4.02, 36.91, 8.18, 11.47)]

South Sudan has the highest growth rate 4.02.

In [59]:
%%sql
SELECT *
    FROM facts
    WHERE migration_rate = (SELECT MAX(migration_rate) FROM facts
                            WHERE name <> 'World')

Done.


[(141, 'qa', 'Qatar', 11586, 11586, 0, 2194817, 3.07, 9.84, 1.53, 22.39)]

Qatar from our output has the highest migration rate

In [60]:
%%sql
SELECT *
    FROM facts
WHERE population * population_growth/100 = (SELECT MAX(population * population_growth/100)
                        FROM facts
                       WHERE name <> 'World');

Done.


[(77, 'in', 'India', 3287263, 2973193, 314070, 1251695584, 1.22, 19.55, 7.32, 0.04)]

India has the highest probablity of adding to their population next year based on the ratio between their population and population growth.

### Water to Land Ratios

In [53]:
%%sql
SELECT name, CAST(area_water AS FLOAT)/CAST(area AS FLOAT) AS ratio_water_to_land
    FROM facts
    ORDER BY ratio_water_to_land DESC;

Done.


[('British Indian Ocean Territory', 0.9988970588235294),
 ('Virgin Islands', 0.818848167539267),
 ('Puerto Rico', 0.35682691610470596),
 ('Bahamas, The', 0.2788184438040346),
 ('Guinea-Bissau', 0.2215916955017301),
 ('Malawi', 0.20596873839505755),
 ('Netherlands', 0.1841465469513516),
 ('Uganda', 0.18228661040997685),
 ('Eritrea', 0.141156462585034),
 ('Liberia', 0.13512736937567904),
 ('Bangladesh', 0.12319816785666172),
 ('Gambia, The', 0.10442477876106195),
 ('Taiwan', 0.10339077265147303),
 ('Finland', 0.10152449393011874),
 ('India', 0.09554148846624076),
 ('Ethiopia', 0.09444897219958345),
 ('Canada', 0.08925312504068737),
 ('Sweden', 0.08874182480373977),
 ('Colombia', 0.08798763730233293),
 ('Brunei', 0.08673026886383348),
 ('Guyana', 0.08429122338569747),
 ('French Polynesia', 0.08159347252219823),
 ('Nicaragua', 0.07961954437370561),
 ('Burundi', 0.07725476104922745),
 ('Iran', 0.07074405637682435),
 ('United States', 0.0676433279822524),
 ('Tanzania', 0.06492135543122558),
 ('Vietnam', 0.06382657528456266),
 ('Rwanda', 0.06340648492672185),
 ('Estonia', 0.0627929601132042),
 ('Norway', 0.06028375365192309),
 ('Armenia', 0.05177688867968934),
 ('Tunisia', 0.05042479066071756),
 ('Uzbekistan', 0.04917299955297273),
 ('Indonesia', 0.04882994525270547),
 ('Suriname', 0.047735319252838485),
 ('Albania', 0.046959788507026576),
 ('Ghana', 0.04611521257016849),
 ('Azerbaijan', 0.045854503464203236),
 ('Togo', 0.04226468257462358),
 ('Russia', 0.04213883509193518),
 ('Kyrgyzstan', 0.04075998619661817),
 ('Tonga', 0.040160642570281124),
 ('Ukraine', 0.04012923535746831),
 ('Lithuania', 0.04012251148545176),
 ('West Bank', 0.03754266211604096),
 ('Gabon', 0.037359853848251745),
 ('Turkmenistan', 0.0372259782831387),
 ('Hungary', 0.036763125080620886),
 ('Latvia', 0.0362290792549815),
 ('Romania', 0.035655708478927475),
 ('Japan', 0.03553709167405369),
 ('Burma', 0.03409806408130327),
 ('Congo, Democratic Republic of the', 0.033183246064367226),
 ('Venezuela', 0.032892933501452774),
 ('Pakistan', 0.03167963622431996),
 ('Hong Kong', 0.0315884476534296),
 ('Solomon Islands', 0.031492248062015504),
 ('Switzerland', 0.031010005572110377),
 ('Moldova', 0.02835957578801217),
 ('China', 0.02819121888598056),
 ('Korea, South', 0.02807862013638187),
 ('Poland', 0.026960039656523337),
 ('Iceland', 0.02669902912621359),
 ('Montenegro', 0.026064291920069503),
 ('Nepal', 0.026022380606192375),
 ('Botswana', 0.025785158063018926),
 ('Laos', 0.02533783783783784),
 ('Cambodia', 0.024967547711768442),
 ('Italy', 0.023893276697418198),
 ('Ecuador', 0.023698604533063432),
 ('Germany', 0.023387914470256736),
 ('Paraguay', 0.023232829832428605),
 ('Belarus', 0.022639691714836225),
 ('Papua New Guinea', 0.021562527007173104),
 ('Bulgaria', 0.02155502845444133),
 ('Senegal', 0.021309258750927706),
 ('Israel', 0.02118440057775638),
 ('Czech Republic', 0.020540910647038686),
 ('Ireland', 0.019780000853812988),
 ('Kenya', 0.019344656053841794),
 ('Slovakia', 0.01896604466197614),
 ('Brazil', 0.01851036371343989),
 ('Tajikistan', 0.017973629424011102),
 ('Turkey', 0.017777789121983967),
 ('Benin', 0.017758519649801994),
 ('Austria', 0.017002301152961096),
 ('Lebanon', 0.016346153846153847),
 ('Mozambique', 0.016262603517726237),
 ('Chile', 0.016254420699852664),
 ('Saint Lucia', 0.016233766233766232),
 ('Somalia', 0.016184249525999086),
 ('New Caledonia', 0.016150740242261104),
 ('Mali', 0.016128147899680048),
 ('Guatemala', 0.015887738890062356),
 ('Argentina', 0.015720759602934828),
 ('Denmark', 0.01531535712628208),
 ('El Salvador', 0.01520840264245996),
 ('Sri Lanka', 0.014936747447035513),
 ('Jamaica', 0.014557365116913839),
 ('Singapore', 0.014347202295552367),
 ('Panama', 0.014319809069212411),
 ('Nigeria', 0.014072797498939128),
 ('Bolivia', 0.01390885150935616),
 ("Cote d'Ivoire", 0.013831044181813107),
 ('Spain', 0.012644201278271365),
 ('Zambia', 0.012250570674631752),
 ('Croatia', 0.01095522493550553),
 ('Macedonia', 0.010889433360556916),


British Indian Ocean Territory has the highest water ratio with percentage about 99.9%
Virgin Islands has water area than land area with percentage about 81.9%.

In [37]:
%%sql
SELECT *
    FROM facts
    WHERE area_water > area_land;

Done.


[(228, 'io', 'British Indian Ocean Territory', 54400, 60, 54340, None, None, None, None, None),
 (247, 'vq', 'Virgin Islands', 1910, 346, 1564, 103574, 0.59, 10.31, 8.54, 7.67)]

### Birth rate VS Death rate

Let's look at the table again.

In [38]:
%%sql
SELECT *
    FROM facts;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46),
 (6, 'ac', 'Antigua and Barbuda', 442, 442, 0, 92436, 1.24, 15.85, 5.69, 2.21),
 (7, 'ar', 'Argentina', 2780400, 2736690, 43710, 43431886, 0.93, 16.64, 7.33, 0.0),
 (8, 'am', 'Armenia', 29743, 28203, 1540, 3056382, 0.15, 13.61, 9.34, 5.8),
 (9, 'as', 'Australia', 7741220, 7682300, 58920, 22751014, 1.07, 12.15, 7.14, 5.65),
 (10, 'au', 'Austria', 83871, 82445, 1426, 8665550, 0.55, 9.41, 9.42, 5.56),
 (11, 'aj', 'Azerbaijan', 86600, 82629, 3971, 9780780, 0.96, 16.64, 7.07, 0.0),
 (12, 'bf', 'Bahamas, The', 13880, 10010, 3870, 324597, 0.85, 15.5, 7.05, 0.0),
 (13, 'ba', 'Bahrain', 760, 760, 0, 1346613, 2.41, 13.66, 2.69, 13.09),
 (14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (15, 'bb', 'Barbados', 430, 430, 0, 290604, 0.31, 11.87, 8.44, 0.3),
 (16, 'bo', 'Belarus', 207600, 202900, 4700, 9589689, 0.2, 10.7, 13.36, 0.7),
 (17, 'be', 'Belgium', 30528, 30278, 250, 11323973, 0.76, 11.41, 9.63, 5.87),
 (18, 'bh', 'Belize', 22966, 22806, 160, 347369, 1.87, 24.68, 5.97, 0.0),
 (19, 'bn', 'Benin', 112622, 110622, 2000, 10448647, 2.78, 36.02, 8.21, 0.0),
 (20, 'bt', 'Bhutan', 38394, 38394, 0, 741919, 1.11, 17.78, 6.69, 0.0),
 (21, 'bl', 'Bolivia', 1098581, 1083301, 15280, 10800882, 1.56, 22.76, 6.52, 0.62),
 (22, 'bk', 'Bosnia and Herzegovina', 51197, 51187, 10, 3867055, 0.13, 8.87, 9.75, 0.38),
 (23, 'bc', 'Botswana', 581730, 566730, 15000, 2182719, 1.21, 20.96, 13.39, 4.56),
 (24, 'br', 'Brazil', 8515770, 8358140, 157630, 204259812, 0.77, 14.46, 6.58, 0.14),
 (25, 'bx', 'Brunei', 5765, 5265, 500, 429646, 1.62, 17.32, 3.52, 2.43),
 (26, 'bu', 'Bulgaria', 110879, 108489, 2390, 7186893, 0.58, 8.92, 14.44, 0.29),
 (27, 'uv', 'Burkina Faso', 274200, 273800, 400, 18931686, 3.03, 42.03, 11.72, 0.0),
 (28, 'bm', 'Burma', 676578, 653508, 23070, 56320206, 1.01, 18.39, 7.96, 0.28),
 (29, 'by', 'Burundi', 27830, 25680, 2150, 10742276, 3.28, 42.01, 9.27, 0.0),
 (30, 'cb', 'Cambodia', 181035, 176515, 4520, 15708756, 1.58, 23.83, 7.68, 0.32),
 (31, 'cm', 'Cameroon', 475440, 472710, 2730, 23739218, 2.59, 36.17, 10.11, 0.15),
 (32, 'ca', 'Canada', 9984670, 9093507, 891163, 35099836, 0.75, 10.28, 8.42, 5.66),
 (33, 'cv', 'Cabo Verde', 4033, 4033, 0, 545993, 1.36, 20.33, 6.11, 0.63),
 (34, 'ct', 'Central African Republic', 622984, 622984, 0, 5391539, 2.13, 35.08, 13.8, 0.0),
 (35, 'cd', 'Chad', None, 1259200, 24800, 11631456, 1.89, 36.6, 14.28, 3.45),
 (36, 'ci', 'Chile', 756102, 743812, 12290, 17508260, 0.82, 13.83, 6.0, 0.34),
 (37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44),
 (38, 'co', 'Colombia', 1138910, 1038700, 100210, 46736728, 1.04, 16.47, 5.4, 0.64),
 (39, 'cn', 'Comoros', 2235, 2235, 0, 780971, 1.77, 27.84, 7.57, 2.53),
 (40, 'cg', 'Congo, Democratic Republic of the', 2344858, 2267048, 77810, 79375136, 2.45, 34.88, 10.07, 0.27),
 (41, 'cf', 'Congo, Republic of the', 342000, 341500, 500, 4755097, 2.0, 35.85, 10.0, 5.9),
 (42, 'cs', 'Costa Rica', 51100, 51060, 40, 4814144, 1.22, 15.91, 4.55, 0.83),
 (43, 'iv', "Cote d'Ivoire", 322463, 318003, 4460, 23295302, 1.91, 28.67, 9.55, 0.0),
 (44, 'hr', 'Croatia', 56594, 55974, 620, 4464844, 0.13, 9.45, 12.18, 1.39),
 (45, 'cu', 'Cuba', 110860, 109820, 1040, 11031433, 0.15, 9.9, 7.72, 3.66),
 (46, 'cy', 'Cyprus', 9251, 9241, 10, 1189197, 1.43, 11.41, 6.62, 9.48),
 (47, 'ez', 'Czech Republic', 78867, 77247, 1620, 10644842, 0.16, 9.63, 10.34, 2.33),
 (48, 'da', 'Denmark', 43094, 42434, 660, 5581503, 0.22, 10.27, 10.25, 2.2),
 (49, 'dj', 'Djibouti', 23200, 23180, 20, 828324, 2.2, 23.65, 7.73, 6.06),
 (50, 'do', 'Dominica', 751, 751, 0, 73607, 0.21, 15.41, 7.91, 

In [54]:
%%sql 
SELECT * 
    FROM facts
    WHERE birth_rate == (SELECT MAX(birth_rate)
                            FROM facts);

Done.


[(128, 'ng', 'Niger', None, 1266700, 300, 18045729, 3.25, 45.45, 12.42, 0.56)]

Niger has the maximum birth_rate 45.45 it's population 18045729 sharing area 1266700.

In [61]:
%%sql
SELECT *
    FROM facts
    WHERE death_rate > birth_rate AND name <> 'World'
    ORDER BY death_rate DESC;

Done.


[(183, 'up', 'Ukraine', 603550, 579330, 24220, 44429471, 0.6, 10.72, 14.46, 2.25),
 (26, 'bu', 'Bulgaria', 110879, 108489, 2390, 7186893, 0.58, 8.92, 14.44, 0.29),
 (96, 'lg', 'Latvia', 64589, 62249, 2340, 1986705, 1.06, 10.0, 14.31, 6.26),
 (102, 'lh', 'Lithuania', 65300, 62680, 2620, 2884433, 1.04, 10.1, 14.27, 6.27),
 (143, 'rs', 'Russia', 17098242, 16377742, 720500, 142423773, 0.04, 11.6, 13.69, 1.69),
 (153, 'ri', 'Serbia', 77474, 77474, 0, 7176794, 0.46, 9.08, 13.66, 0.0),
 (16, 'bo', 'Belarus', 207600, 202900, 4700, 9589689, 0.2, 10.7, 13.36, 0.7),
 (75, 'hu', 'Hungary', 93028, 89608, 3420, 9897541, 0.22, 9.16, 12.73, 1.33),
 (116, 'md', 'Moldova', 33851, 32891, 960, 3546847, 1.03, 12.0, 12.59, 9.67),
 (57, 'en', 'Estonia', 45228, 42388, 2840, 1265420, 0.55, 10.51, 12.4, 3.6),
 (44, 'hr', 'Croatia', 56594, 55974, 620, 4464844, 0.13, 9.45, 12.18, 1.39),
 (142, 'ro', 'Romania', 238391, 229891, 8500, 21666350, 0.3, 9.14, 11.9, 0.24),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (158, 'si', 'Slovenia', 20273, 20151, 122, 1983412, 0.26, 8.42, 11.37, 0.37),
 (67, 'gr', 'Greece', 131957, 130647, 1310, 10775643, 0.01, 8.66, 11.09, 2.32),
 (140, 'po', 'Portugal', 92090, 91470, 620, 10825309, 0.09, 9.27, 11.02, 2.67),
 (47, 'ez', 'Czech Republic', 78867, 77247, 1620, 10644842, 0.16, 9.63, 10.34, 2.33),
 (83, 'it', 'Italy', 301340, 294140, 7200, 61855120, 0.27, 8.74, 10.19, 4.1),
 (139, 'pl', 'Poland', 312685, 304255, 8430, 38562189, 0.09, 9.74, 10.19, 0.46),
 (22, 'bk', 'Bosnia and Herzegovina', 51197, 51187, 10, 3867055, 0.13, 8.87, 9.75, 0.38),
 (214, 'sb', 'Saint Pierre and Miquelon', 242, 242, 0, 5657, 1.08, 7.42, 9.72, 8.49),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (10, 'au', 'Austria', 83871, 82445, 1426, 8665550, 0.55, 9.41, 9.42, 5.56),
 (117, 'mn', 'Monaco', 2, 2, 0, 30535, 0.12, 6.65, 9.24, 3.83)]

Ukraine has the highest death_rate around 14.46.

### Area/Population Ratio

In [57]:
%%sql
SELECT name, CAST(population AS FLOAT)/CAST(area AS FLOAT) AS ratio_pop_area
    FROM facts
    
    ORDER BY ratio_pop_area DESC;

Done.


[('Macau', 21168.964285714286),
 ('Monaco', 15267.5),
 ('Singapore', 8141.279770444763),
 ('Hong Kong', 6445.041516245487),
 ('Gaza Strip', 5191.819444444444),
 ('Gibraltar', 4876.333333333333),
 ('Bahrain', 1771.8592105263158),
 ('Maldives', 1319.6409395973155),
 ('Malta', 1310.01582278481),
 ('Bermuda', 1299.9259259259259),
 ('Sint Maarten', 1167.3235294117646),
 ('Bangladesh', 1138.0691432035565),
 ('Guernsey', 847.1794871794872),
 ('Jersey', 838.7413793103449),
 ('Barbados', 675.8232558139534),
 ('Mauritius', 656.7779411764706),
 ('Taiwan', 650.7817120622568),
 ('Aruba', 623.1222222222223),
 ('Lebanon', 594.6827884615385),
 ('Saint Martin', 588.0370370370371),
 ('San Marino', 541.311475409836),
 ('Korea, South', 492.53104693140796),
 ('Rwanda', 480.74010934771053),
 ('West Bank', 475.3184300341297),
 ('Nauru', 454.2857142857143),
 ('Tuvalu', 418.03846153846155),
 ('Netherlands', 407.96052283176465),
 ('Marshall Islands', 398.8453038674033),
 ('Israel', 387.5452094366875),
 ('Burundi', 385.99626302551206),
 ('India', 380.7713541630226),
 ('Belgium', 370.9372707023061),
 ('Haiti', 364.325009009009),
 ('Comoros', 349.42774049217),
 ('Philippines', 336.6612533333333),
 ('Sri Lanka', 336.12998018594726),
 ('Japan', 335.8418136353413),
 ('Curacao', 334.2477477477477),
 ('Grenada', 321.7848837209302),
 ('Guam', 297.39889705882354),
 ('El Salvador', 291.8753861508483),
 ('Vietnam', 284.8610700160019),
 ('American Samoa', 273.08040201005025),
 ('Jamaica', 268.4205258848148),
 ('Saint Lucia', 266.10714285714283),
 ('Saint Vincent and the Grenadines', 263.82262210796915),
 ('United Kingdom', 263.0771396904889),
 ('Puerto Rico', 260.92067290261764),
 ('Pakistan', 250.07800199724906),
 ('Trinidad and Tobago', 238.3703198127925),
 ('Liechtenstein', 235.15),
 ('Germany', 226.4689795026637),
 ('British Virgin Islands', 221.5496688741722),
 ('Luxembourg', 220.51508120649652),
 ('Dominican Republic', 215.3021573864804),
 ('Nepal', 214.3707747603291),
 ('Cayman Islands', 212.46969696969697),
 ('Antigua and Barbuda', 209.13122171945702),
 ('Korea, North', 207.26414076888616),
 ('Italy', 205.26687462666754),
 ('Seychelles', 203.14285714285714),
 ('Sao Tome and Principe', 201.25103734439833),
 ('Saint Kitts and Nevis', 198.98850574712642),
 ('Switzerland', 196.76405746541658),
 ('Nigeria', 196.5450805830035),
 ('Qatar', 189.43699292249266),
 ('Andorra', 182.86324786324786),
 ('Anguilla', 180.41758241758242),
 ('Gambia, The', 174.13353982300885),
 ('Kosovo', 171.8545972260494),
 ('Kuwait', 156.50095409136827),
 ('Uganda', 153.92487906471177),
 ('Isle of Man', 153.0506993006993),
 ('Malawi', 151.62129063839842),
 ('Micronesia, Federated States of', 149.8803418803419),
 ('Tonga', 142.57161981258366),
 ('China', 142.4915168970174),
 ('Guatemala', 137.01107549890256),
 ('Cabo Verde', 135.3813538308951),
 ('Czech Republic', 134.97206689743493),
 ('Indonesia', 134.41029125224657),
 ('Togo', 132.99846790525666),
 ('Thailand', 132.47662340193327),
 ('Kiribati', 130.34648581997533),
 ('Denmark', 129.51926022184063),
 ('Cyprus', 128.54794076316074),
 ('Akrotiri', 127.64227642276423),
 ('Poland', 123.3259958104802),
 ('Dhekelia', 120.76923076923077),
 ('European Union', 118.83823161491145),
 ('Portugal', 117.5514062330329),
 ('Azerbaijan', 112.94203233256351),
 ('Northern Mariana Islands', 112.8103448275862),
 ('Tokelau', 111.41666666666667),
 ('Slovakia', 111.04368308351178),
 ('Ghana', 110.37319364616216),
 ('Wallis and Futuna', 109.95070422535211),
 ('Hungary', 106.3931396998753),
 ('Albania', 105.37352163628775),
 ('Moldova', 104.77820448435793),
 ('France', 103.37630106197412),
 ('Austria', 103.31997949231558),
 ('Armenia', 102.75970816662745),
 ('Turkey', 101.35033220089795),
 ('Cuba', 99.50778459318059),
 ('Dominica', 98.01198402130493),
 ('Slovenia', 97.8351501997731),
 ('Spain', 95.26907810119319),
 ('Costa Rica', 94.21025440313112),
 ('Benin', 92.77625153167232),
 ('Serbia', 92.63487105351473),
 ('Malaysia', 92.50909664177634),
 ('Syria', 92.1527

Macau has the highest population/area ratio and the area of land in square kilometer is so small posibly compared to other countries.